flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/BigQueryDynamicTableFactory.java

```diff
@@ -102,11 +113,35 @@ public DynamicTableSource createDynamicTableSource(Context context) {
             config = config.withTestingServices(testingServices);
         }
 
+        // Create an unbounded source.
+        if (config.isUnboundedEnabled()) {
+            return new BigQueryDynamicTableSource(
+                    config.toBigQueryReadOptions(),
+                    context.getPhysicalRowDataType(),
+                    Boundedness.CONTINUOUS_UNBOUNDED);
+        }
+
         return new BigQueryDynamicTableSource(
                 config.toBigQueryReadOptions(), context.getPhysicalRowDataType());
     }
```

Comment:
Combine into one method, and use ternary operator to distinguish between the two modes.

Reply:
Fixed.